prov/shm/src/smr_msg.c

```diff
@@ -70,7 +70,9 @@ static struct smr_rx_entry *smr_get_recv_entry(struct smr_ep *ep,
 	entry->context = context;
 	entry->err = 0;
 	entry->flags = smr_convert_rx_flags(flags);
-	entry->addr = ep->util_ep.caps & FI_DIRECTED_RECV ? addr : FI_ADDR_UNSPEC;
+	entry->addr = ep->util_ep.caps & FI_DIRECTED_RECV ?
+		*((fi_addr_t *) ofi_av_get_addr(ep->util_ep.av, addr)) :
```

Comment:
Same here.  If shm is storing indices, or int, then ofi_av_get_addr should be cast to (int *) or (u64 *) or (u32 *), not fi_addr_t *, which the commit comment indicates is what is NOT being stored.

Reply:
You're right. That is confusing. I kept it as an fi_addr because that is in the protocol (see struct smr_msg_hdr) and because it uses FI_ADDR_UNSPEC for invalid/unmapped entries. But I can change that to a u64 and check for -1 if that helps? Would you prefer it that way?

Reply:
I think converting it to uint64 or int64 would be less confusing, if these are just indexes and require a conversion to get a usable fi_addr_t value.